compiler/rustc_next_trait_solver/src/solve/assembly/structural_traits.rs

```diff
@@ -764,9 +773,12 @@ pub(in crate::solve) fn const_conditions_for_destruct<I: Interner>(
         | ty::Coroutine(_, _)
         | ty::CoroutineWitness(_, _) => Err(NoSolution),
 
-        ty::Dynamic(..) | ty::Param(_) | ty::Alias(..) | ty::Placeholder(_) | ty::Foreign(_) => {
-            Err(NoSolution)
-        }
+        ty::Dynamic(..)
+        | ty::Param(_)
+        | ty::Alias(..)
+        | ty::Placeholder(_)
+        | ty::Foreign(_)
+        | ty::UnsafeBinder(_) => Err(NoSolution),
 
```

Comment:
Should probably have a FIXME? If it's not acting like ManuallyDrop outside of const contexts, it should be droppable in const contexts if its bound ty is droppable in const contexts.

Reply:
Yeah, this is hard to implement in practice b/c we have to instantiate the binder, so I'll split it and leave a FIXME.